pkg/proctree/fileinfo.go

```diff
@@ -0,0 +1,287 @@
+package proctree
+
+import (
+	"sync"
+	"time"
+
+	ch "github.com/aquasecurity/tracee/pkg/changelog"
+)
+
+// FileInfoFeed allows external packages to set/get multiple values of a task at once.
+type FileInfoFeed struct {
+	Name      string
+	Path      string
+	Dev       int
+	Ctime     int
+	Inode     int
+	InodeMode int
+}
+
+//
+// File Info
+//
+
+// FileInfo represents a file.
+type FileInfo struct {
+	name      *ch.Changelog[string] // file name
```

Comment:
I'm assuming from this exec event:

```
Exec event received:
Task Hash=2499140067
Path Name=/usr/bin/bash
Command Path=/opt/visual-studio-code/resources/app/out/vs/base/node/cpuUsage.sh
Device=271581188
Ctime=1693757533306530358
Inode=53615542
Inode Mode=33261
Interpreter Path=/usr/lib/ld-linux-x86-64.so.2
Interpreter Device=271581188
Interpreter Ctime=1693757532899867842
Interpreter Inode=53611806
Interp=/bin/bash
Stdin Path=UNIX-STREAM
Stdin Type=49152
Kernel Invoked=0
```

You are basically saying that the "Path Name" isn't important because you can pick from Command Path. But if there are many different combinations. For example.. in the case above, the path name is the same as Interp (because its bash executing a script)... so picking Interp would already be enough. But in the case:

```
Exec event received:
Task Hash=2499140067
Path Name=/usr/bin/bash
Command Path=/bin/sh
Device=271581188
Ctime=1693757533306530358
Inode=53615542
Inode Mode=33261
Interpreter Path=/usr/lib/ld-linux-x86-64.so.2
Interpreter Device=271581188
Interpreter Ctime=1693757532899867842
Interpreter Inode=53611806
Interp=/bin/sh
Stdin Path=UNIX-STREAM
Stdin Type=49152
Kernel Invoked=0
```

Then you will see that the "command path" is "/bin/sh" while the path name is "/usr/bin/bash" (and Interp is now same as command path and not path name".